src/vs/workbench/parts/files/electron-browser/views/explorerView.ts

```diff
@@ -342,12 +342,13 @@ export class ExplorerView extends ViewletPanel {
 
 				if (e.browserEvent instanceof MouseEvent) {
 					isDoubleClick = e.browserEvent.detail === 2;
+					isMiddleClick = e.browserEvent.button === 1;
+					const isLeftButton = e.browserEvent.button === 0;
 
-					if (!this.tree.openOnSingleClick && !isDoubleClick) {
+					if (isLeftButton && !this.tree.openOnSingleClick && !isDoubleClick) {
```

Comment:
Same as above: distinguishing between single/double click should only be relevant when using the left button.

Unfortunately, @isidorn still hasn't adopted the open event. He has an issue for that.